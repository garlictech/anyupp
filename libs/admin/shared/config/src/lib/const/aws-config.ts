import { environment } from '../environments/environment';

export const AWS_CONFIG = {
  aws_appsync_graphqlEndpoint: environment.config.GraphqlApiUrl,
  api_key: environment.config.GraphqlApiKey,
  aws_appsync_region: 'eu-west-1',
  aws_project_region: 'eu-west-1',
  aws_cognito_region: 'eu-west-1',
  // aws_cognito_identity_pool_id: environment.config.IdentityPoolId,
  aws_user_pools_id: environment.config.UserPoolId,
  aws_user_pools_web_client_id: environment.config.UserPoolClientId,
  oauth: {
    domain: environment.config.UserPoolDomain,
    scope: ['openid'],
    redirectSignIn: `${environment.config.AdminSiteUrl}/admin/dashboard`,
    redirectSignOut: `${environment.config.AdminSiteUrl}/auth/login`,
    responseType: 'code',
  },
  // aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  // federationTarget: 'COGNITO_USER_AND_IDENTITY_POOLS', ???
};
