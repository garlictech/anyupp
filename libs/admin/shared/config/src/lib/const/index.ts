export { AWS_CONFIG } from './aws-config';
