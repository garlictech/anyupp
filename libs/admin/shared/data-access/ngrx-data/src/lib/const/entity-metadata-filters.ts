import * as CrudApi from '@bgap/crud-gql/api';

// Return true, if no name filter has been specified
export const localizedItemFilter =
  <T>(
    fieldName: keyof T,
    lang: string,
    toFind?: CrudApi.LocalizedItem | null,
  ) =>
  (item: T) =>
    toFind?.[<keyof CrudApi.LocalizedItem>lang]
      ? -1 <
        (
          (<CrudApi.LocalizedItem>item?.[fieldName])?.[
            <keyof CrudApi.LocalizedItem>lang
          ]?.toLocaleLowerCase() || ''
        ).indexOf(
          (
            toFind?.[<keyof CrudApi.LocalizedItem>lang] || ''
          ).toLocaleLowerCase(),
        )
      : true;

export const simpleFilter =
  <T>(property: keyof T, fallback: boolean, toFind?: unknown) =>
  (item: T) =>
    toFind ? <unknown>item[property] === toFind : fallback;
