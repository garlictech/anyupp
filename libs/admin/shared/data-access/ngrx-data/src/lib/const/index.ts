export * from './entity-metadata-filters';
export * from './entity-metadata';
export * from './pagination';
