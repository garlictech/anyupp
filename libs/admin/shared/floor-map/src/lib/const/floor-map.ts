import { EUnitMapObjectType } from '@bgap/shared/types';

export const FLOOR_MAP_CONFIG = {
  grid: 10,
  gridDivider: 5,
  backgroundColor: '#f8f8f8',
  lineStroke: '#ebebeb',
  strokeWidth: 1,
  cornerRadius: 2,
  tableFill: 'rgba(150, 111, 51, 0.7)',
  tableStroke: '#694d23',
  seatFill: 'rgba(67, 42, 4, 0.7)',
  seatStroke: '#32230b',
  barFill: 'rgba(0, 93, 127, 0.7)',
  barStroke: '#003e54',
  wallFill: 'rgba(136, 136, 136, 0.7)',
};

export const FLOOR_MAP_TEXT_CONFIG = {
  fontFamily: 'Arial',
  fontSize: 14,
  fill: '#fff',
  textAlign: 'center',
  originX: 'center',
  originY: 'center',
  selectable: false,
};

export const FLOOR_MAP_COMMON_BG_OPTIONS = {
  strokeWidth: FLOOR_MAP_CONFIG.strokeWidth,
  originX: 'center',
  originY: 'center',
  selectable: false,
  type: 'bg',
};

export const FLOOR_MAP_GRID_OPTIONS = {
  stroke: FLOOR_MAP_CONFIG.lineStroke,
  selectable: false,
  type: 'line',
  hoverCursor: 'default',
};

export const FLOOR_MAP_ENABLED_GROUP_OPTIONS = {
  centeredRotation: true,
  snapAngle: 15,
  selectable: true,
};

export const FLOOR_MAP_DISABLED_GROUP_OPTIONS = {
  selectable: false,
  hoverCursor: 'default',
};

export const FLOOR_MAP_CIRCLE_CONTROLS = {
  tl: true,
  tr: true,
  bl: true,
  br: true,
  mtr: false,
  mb: false,
  ml: false,
  mr: false,
  mt: false,
};

export const FLOOR_MAP_OBJECT_COMMON_DEFAULTS = {

    x: 10,
    y: 10,
    c: '',
    a: 0,

}

export const FLOOR_MAP_OBJECT_DEFAULTS = {
  [EUnitMapObjectType.TABLE_RECTANGLE]: {
    w: 150,
    h: 60,
  },
  [EUnitMapObjectType.TABLE_CIRCLE]: {
    r: 30,
  },
  [EUnitMapObjectType.SEAT_RECTANGLE]: {
    w: 30,
    h: 30,
  },
  [EUnitMapObjectType.SEAT_CIRCLE]: {
    r: 15,
  },
  [EUnitMapObjectType.COUNTER]: {
    w: 180,
    h: 60,
  },
  [EUnitMapObjectType.WALL]: {
    w: 10,
    h: 60,
  },
  [EUnitMapObjectType.LABEL]: {
    w: 150,
    h: 60,
    c: '..........',
  },
};
