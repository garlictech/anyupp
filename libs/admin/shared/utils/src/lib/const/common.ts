import * as CrudApi from '@bgap/crud-gql/api';
import { PaymentMode } from '@bgap/crud-gql/api';
import { IAllergen } from '@bgap/shared/types';

export const DEFAULT_LANG = 'en-US';

export const WEEKLY_VARIANT_AVAILABILITY = {
  MONDAY: 'common.days.mon',
  TUESDAY: 'common.days.tue',
  WEDNESDAY: 'common.days.wed',
  THURSDAY: 'common.days.thu',
  FRIDAY: 'common.days.fri',
  SATURDAY: 'common.days.sat',
  SUNDAY: 'common.days.sun',
};

export const TIME_FORMAT_PATTERN = '^([0-1][0-9]|2[0-3]):[0-5][0-9]$';

export const MENU_ROLES = {
  DASHBOARD: [
    CrudApi.Role.superuser,
    CrudApi.Role.chainadmin,
    CrudApi.Role.groupadmin,
    CrudApi.Role.unitadmin,
    CrudApi.Role.staff,
  ],
  PRODUCTS: [
    CrudApi.Role.superuser,
    CrudApi.Role.chainadmin,
    CrudApi.Role.groupadmin,
    CrudApi.Role.unitadmin,
  ],
  PRODUCT_CATEGORIES: [CrudApi.Role.superuser, CrudApi.Role.chainadmin],
  MODIFIERS_AND_EXTRAS: [CrudApi.Role.superuser, CrudApi.Role.chainadmin],
  UNITS: [
    CrudApi.Role.superuser,
    CrudApi.Role.chainadmin,
    CrudApi.Role.groupadmin,
    CrudApi.Role.unitadmin,
  ],
  GROUPS: [
    CrudApi.Role.superuser,
    CrudApi.Role.chainadmin,
    CrudApi.Role.groupadmin,
  ],
  CHAINS: [CrudApi.Role.superuser, CrudApi.Role.chainadmin],
  USERS: [CrudApi.Role.superuser],
  ADMINS: [
    CrudApi.Role.superuser,
    CrudApi.Role.chainadmin,
    CrudApi.Role.groupadmin,
    CrudApi.Role.unitadmin,
  ],
  ROLE_CONTEXTS: [
    CrudApi.Role.superuser,
    CrudApi.Role.chainadmin,
    CrudApi.Role.groupadmin,
    CrudApi.Role.unitadmin,
  ],
};

export const PAYMENT_MODES: PaymentMode[] = [
  {
    type: CrudApi.PaymentType.cash,
    caption: 'common.paymentModes.cash',
    method: CrudApi.PaymentMethod.cash,
  },
  {
    type: CrudApi.PaymentType.card,
    caption: 'common.paymentModes.card',
    method: CrudApi.PaymentMethod.card,
  },
  {
    type: CrudApi.PaymentType.googlepay,
    caption: 'common.paymentModes.googlepay',
    method: CrudApi.PaymentMethod.inapp,
  },
  {
    type: CrudApi.PaymentType.applepay,
    caption: 'common.paymentModes.applepay',
    method: CrudApi.PaymentMethod.inapp,
  },
  {
    type: CrudApi.PaymentType.stripe,
    caption: 'common.paymentModes.stripe',
    method: CrudApi.PaymentMethod.inapp,
  },
  {
    type: CrudApi.PaymentType.simple,
    caption: 'common.paymentModes.simple',
    method: CrudApi.PaymentMethod.inapp,
  },
];

export const SERVING_MODES = [
  CrudApi.ServingMode.inplace,
  CrudApi.ServingMode.takeaway,
];

export const ORDER_MODES = [
  CrudApi.OrderMode.instant,
  CrudApi.OrderMode.pickup,
];

export const DEFAULT_LANE_COLOR = '#5B7AFF';

export const ALLERGENS: IAllergen[] = [
  {
    id: 'gluten',
    idx: 1,
  },
  {
    id: 'crustaceans',
    idx: 2,
  },
  {
    id: 'egg',
    idx: 3,
  },
  {
    id: 'fish',
    idx: 4,
  },
  {
    id: 'peanut',
    idx: 5,
  },
  {
    id: 'milk',
    idx: 6,
  },
  {
    id: 'soya',
    idx: 7,
  },
  {
    id: 'treenuts',
    idx: 8,
  },
  {
    id: 'sulphites',
    idx: 9,
  },
  {
    id: 'mustard',
    idx: 10,
  },
  {
    id: 'celery',
    idx: 11,
  },
  {
    id: 'sesame',
    idx: 12,
  },
  {
    id: 'lupin',
    idx: 13,
  },
  {
    id: 'molluscs',
    idx: 14,
  },
];
