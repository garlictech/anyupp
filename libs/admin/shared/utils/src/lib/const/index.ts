export * from './common';
export * from './floor-map';
