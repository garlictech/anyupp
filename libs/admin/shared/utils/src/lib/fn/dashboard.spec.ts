import * as CrudApi from '@bgap/crud-gql/api';
import { orderFixture as ofx } from '@bgap/shared/fixtures';

import { timezoneBudapest } from '../const';
import {
  calculatePaymentMethodSums,
  calculateProductMix,
  calculateUnpayCategoryStat,
  dailySalesPerPaymentMethodOrderAmounts,
  dailySalesPerTypeOrderAmounts,
  getDailyOrdersSum,
  hourlyBreakdownOrderAmounts,
  unpayCategoryTableData,
} from './dashboard';

const singleOrder = [ofx.convertInputToOrder(ofx.historySuccessCardOrderInput)];
const successHistoryOrders = [
  ofx.convertInputToOrder(ofx.historySuccessCardOrderInput),
  ofx.convertInputToOrder(ofx.historySuccessCashOrderInput),
  ofx.convertInputToOrder(ofx.historySuccessStripeOrderInput),
];
const failedHistoryOrders = [
  ofx.convertInputToOrder(ofx.historyFailedCardStaffMealOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCardManagerMealOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCardMarketingPromoOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCardErrorCookedOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCardErrorNoCookedOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCardPaymentModeChangedOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCardOtherOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCardDeliveryOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCardCouponOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCardEventOrderInput),

  ofx.convertInputToOrder(ofx.historyFailedCashStaffMealOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCashManagerMealOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCashMarketingPromoOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCashErrorCookedOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCashErrorNoCookedOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCashPaymentModeChangedOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCashOtherOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCashDeliveryOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCashCouponOrderInput),
  ofx.convertInputToOrder(ofx.historyFailedCashEventOrderInput),
];

const paymentMethods = [
  CrudApi.PaymentMethod.card,
  CrudApi.PaymentMethod.cash,
  CrudApi.PaymentMethod.inapp,
];

describe('Dashboard pure function tests', () => {
  describe('calculatePaymentMethodSums', () => {
    it('should calculate 1 item', () => {
      const result = calculatePaymentMethodSums(
        paymentMethods,
        failedHistoryOrders,
      );

      expect(result).toMatchSnapshot();
    });
  });

  describe('calculateUnpayCategoryStat', () => {
    it('should calculate 1 item', () => {
      const result = calculateUnpayCategoryStat(
        CrudApi.UnpayCategory.manager_meal,
        failedHistoryOrders,
        paymentMethods,
      );

      expect(result).toMatchSnapshot();
    });
  });

  describe('getDailyOrdersSum', () => {
    it('should calculate 1 item', () => {
      const result = getDailyOrdersSum(singleOrder);

      expect(result).toMatchSnapshot();
    });

    it('should calculate success history items', () => {
      const result = getDailyOrdersSum(successHistoryOrders);

      expect(result).toMatchSnapshot();
    });

    it('should calculate failed history items', () => {
      const result = getDailyOrdersSum(failedHistoryOrders);

      expect(result).toMatchSnapshot();
    });
  });

  describe('hourlyBreakdownOrderAmounts', () => {
    it('should calculate 1 item', () => {
      const result = hourlyBreakdownOrderAmounts(timezoneBudapest, singleOrder);

      expect(result).toMatchSnapshot();
    });

    it('should calculate success history items', () => {
      const result = hourlyBreakdownOrderAmounts(
        timezoneBudapest,
        successHistoryOrders,
      );

      expect(result).toMatchSnapshot();
    });

    it('should calculate failed history items', () => {
      const result = hourlyBreakdownOrderAmounts(
        timezoneBudapest,
        failedHistoryOrders,
      );

      expect(result).toMatchSnapshot();
    });
  });

  describe('dailySalesPerTypeOrderAmounts', () => {
    it('should calculate 1 item', () => {
      const result = dailySalesPerTypeOrderAmounts(singleOrder);

      expect(result).toMatchSnapshot();
    });

    it('should calculate success history items', () => {
      const result = dailySalesPerTypeOrderAmounts(successHistoryOrders);

      expect(result).toMatchSnapshot();
    });

    it('should calculate failed history items', () => {
      const result = dailySalesPerTypeOrderAmounts(failedHistoryOrders);

      expect(result).toMatchSnapshot();
    });
  });

  describe('dailySalesPerPaymentMethodOrderAmounts', () => {
    it('should calculate 1 item', () => {
      const result = dailySalesPerPaymentMethodOrderAmounts(singleOrder);

      expect(result).toMatchSnapshot();
    });

    it('should calculate success history items', () => {
      const result =
        dailySalesPerPaymentMethodOrderAmounts(successHistoryOrders);

      expect(result).toMatchSnapshot();
    });

    it('should calculate failed history items', () => {
      const result =
        dailySalesPerPaymentMethodOrderAmounts(failedHistoryOrders);

      expect(result).toMatchSnapshot();
    });
  });

  describe('calculateProductMix', () => {
    it('should calculate 1 item', () => {
      const result = calculateProductMix(singleOrder);

      expect(result).toMatchSnapshot();
    });

    it('should calculate lots of order items', () => {
      const result = calculateProductMix([
        ...successHistoryOrders,
        ...failedHistoryOrders,
      ]);

      expect(result).toMatchSnapshot();
    });
  });

  describe('unpayCategoryTableData', () => {
    it('should calculate hasIncome stats', () => {
      const result = unpayCategoryTableData(
        failedHistoryOrders,
        true,
        paymentMethods,
      );

      expect(result).toMatchSnapshot();
    });

    it('should calculate hasIncome stats', () => {
      const result = unpayCategoryTableData(
        failedHistoryOrders,
        false,
        paymentMethods,
      );

      expect(result).toMatchSnapshot();
    });
  });

  describe('unpayCategoryTableData', () => {
    it('should calculate hasIncome stats', () => {
      const result = unpayCategoryTableData(
        failedHistoryOrders,
        true,
        paymentMethods,
      );
      const expected = [
        {
          category: 'delivery',
          count: 2,
          paymentMethodSums: {
            card: 2982,
            cash: 2982,
            inapp: 0,
          },
          sum: 5964,
          uniqueUsersCount: 1,
        },
        {
          category: 'coupon',
          count: 2,
          paymentMethodSums: {
            card: 2982,
            cash: 2982,
            inapp: 0,
          },
          sum: 5964,
          uniqueUsersCount: 1,
        },
        {
          category: 'event',
          count: 2,
          paymentMethodSums: {
            card: 2982,
            cash: 2982,
            inapp: 0,
          },
          sum: 5964,
          uniqueUsersCount: 1,
        },
        {
          category: 'sum',
          count: 6,
          paymentMethodSums: {
            card: 8946,
            cash: 8946,
            inapp: 0,
          },
          sum: 17892,
          uniqueUsersCount: 1,
        },
      ];
      expect(result).toEqual(expected);
    });

    it('should calculate hasIncome stats', () => {
      const result = unpayCategoryTableData(
        failedHistoryOrders,
        false,
        paymentMethods,
      );
      const expected = [
        {
          category: 'staff_meal',
          count: 2,
          paymentMethodSums: {
            card: 2982,
            cash: 2982,
            inapp: 0,
          },
          sum: 5964,
          uniqueUsersCount: 1,
        },
        {
          category: 'manager_meal',
          count: 2,
          paymentMethodSums: {
            card: 2982,
            cash: 2982,
            inapp: 0,
          },
          sum: 5964,
          uniqueUsersCount: 1,
        },
        {
          category: 'marketing_promo',
          count: 2,
          paymentMethodSums: {
            card: 2982,
            cash: 2982,
            inapp: 0,
          },
          sum: 5964,
          uniqueUsersCount: 1,
        },
        {
          category: 'error_cooked',
          count: 2,
          paymentMethodSums: {
            card: 2982,
            cash: 2982,
            inapp: 0,
          },
          sum: 5964,
          uniqueUsersCount: 1,
        },
        {
          category: 'error_no_cooked',
          count: 2,
          paymentMethodSums: {
            card: 2982,
            cash: 2982,
            inapp: 0,
          },
          sum: 5964,
          uniqueUsersCount: 1,
        },
        {
          category: 'payment_mode_change',
          count: 2,
          paymentMethodSums: {
            card: 2982,
            cash: 2982,
            inapp: 0,
          },
          sum: 5964,
          uniqueUsersCount: 1,
        },
        {
          category: 'other',
          count: 2,
          paymentMethodSums: {
            card: 2982,
            cash: 2982,
            inapp: 0,
          },
          sum: 5964,
          uniqueUsersCount: 1,
        },
        {
          category: 'sum',
          count: 14,
          paymentMethodSums: {
            card: 20874,
            cash: 20874,
            inapp: 0,
          },
          sum: 41748,
          uniqueUsersCount: 1,
        },
      ];
      expect(result).toEqual(expected);
    });
  });
});
