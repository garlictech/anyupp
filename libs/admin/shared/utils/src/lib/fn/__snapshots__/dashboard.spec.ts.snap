// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard pure function tests calculatePaymentMethodSums should calculate 1 item 1`] = `
Object {
  "card": 29820,
  "cash": 29820,
  "inapp": 0,
}
`;

exports[`Dashboard pure function tests calculateProductMix should calculate 1 item 1`] = `
Array [
  Object {
    "components": Array [
      Object {
        "componentId": "test_product_component_id",
        "name": Object {
          "de": "Room temperature",
          "en": "Room temperature",
          "hu": "Szobahőmérsékletű",
        },
        "quantity": 5,
      },
    ],
    "name": Object {
      "de": null,
      "en": "Hamburger",
      "hu": "Hamburger",
    },
    "productId": "test_unit_product_hamburger",
    "productType": "food",
    "quantity": 5,
    "variants": Array [
      Object {
        "name": Object {
          "de": null,
          "en": "glass",
          "hu": "pohár",
        },
        "quantity": 5,
        "variantId": "test_variant_id",
      },
    ],
  },
  Object {
    "components": Array [
      Object {
        "componentId": "test_product_component_id",
        "name": Object {
          "de": "Room temperature",
          "en": "Room temperature",
          "hu": "Szobahőmérsékletű",
        },
        "quantity": 5,
      },
    ],
    "name": Object {
      "de": null,
      "en": "Fanta",
      "hu": "Fanta",
    },
    "productId": "test_unit_product_fanta",
    "productType": "drink",
    "quantity": 5,
    "variants": Array [
      Object {
        "name": Object {
          "de": null,
          "en": "glass",
          "hu": "pohár",
        },
        "quantity": 5,
        "variantId": "test_variant_id",
      },
    ],
  },
]
`;

exports[`Dashboard pure function tests calculateProductMix should calculate lots of order items 1`] = `
Array [
  Object {
    "components": Array [
      Object {
        "componentId": "test_product_component_id",
        "name": Object {
          "de": "Room temperature",
          "en": "Room temperature",
          "hu": "Szobahőmérsékletű",
        },
        "quantity": 115,
      },
    ],
    "name": Object {
      "de": null,
      "en": "Hamburger",
      "hu": "Hamburger",
    },
    "productId": "test_unit_product_hamburger",
    "productType": "food",
    "quantity": 115,
    "variants": Array [
      Object {
        "name": Object {
          "de": null,
          "en": "glass",
          "hu": "pohár",
        },
        "quantity": 115,
        "variantId": "test_variant_id",
      },
    ],
  },
  Object {
    "components": Array [
      Object {
        "componentId": "test_product_component_id",
        "name": Object {
          "de": "Room temperature",
          "en": "Room temperature",
          "hu": "Szobahőmérsékletű",
        },
        "quantity": 115,
      },
    ],
    "name": Object {
      "de": null,
      "en": "Fanta",
      "hu": "Fanta",
    },
    "productId": "test_unit_product_fanta",
    "productType": "drink",
    "quantity": 115,
    "variants": Array [
      Object {
        "name": Object {
          "de": null,
          "en": "glass",
          "hu": "pohár",
        },
        "quantity": 115,
        "variantId": "test_variant_id",
      },
    ],
  },
]
`;

exports[`Dashboard pure function tests calculateUnpayCategoryStat should calculate 1 item 1`] = `
Object {
  "category": "manager_meal",
  "count": 2,
  "paymentMethodSums": Object {
    "card": 2982,
    "cash": 2982,
    "inapp": 0,
  },
  "sum": 5964,
  "uniqueUsersCount": 1,
}
`;

exports[`Dashboard pure function tests dailySalesPerPaymentMethodOrderAmounts should calculate 1 item 1`] = `
Object {
  "card": 2982,
  "cash": 0,
  "inapp": 0,
}
`;

exports[`Dashboard pure function tests dailySalesPerPaymentMethodOrderAmounts should calculate failed history items 1`] = `
Object {
  "card": 29820,
  "cash": 29820,
  "inapp": 0,
}
`;

exports[`Dashboard pure function tests dailySalesPerPaymentMethodOrderAmounts should calculate success history items 1`] = `
Object {
  "card": 2982,
  "cash": 2982,
  "inapp": 2982,
}
`;

exports[`Dashboard pure function tests dailySalesPerTypeOrderAmounts should calculate 1 item 1`] = `
Object {
  "drink": 1491,
  "food": 1491,
  "other": 0,
}
`;

exports[`Dashboard pure function tests dailySalesPerTypeOrderAmounts should calculate failed history items 1`] = `
Object {
  "drink": 29820,
  "food": 29820,
  "other": 0,
}
`;

exports[`Dashboard pure function tests dailySalesPerTypeOrderAmounts should calculate success history items 1`] = `
Object {
  "drink": 4473,
  "food": 4473,
  "other": 0,
}
`;

exports[`Dashboard pure function tests getDailyOrdersSum should calculate 1 item 1`] = `2982`;

exports[`Dashboard pure function tests getDailyOrdersSum should calculate failed history items 1`] = `59640`;

exports[`Dashboard pure function tests getDailyOrdersSum should calculate success history items 1`] = `8946`;

exports[`Dashboard pure function tests hourlyBreakdownOrderAmounts should calculate 1 item 1`] = `
Object {
  "drink": Array [
    0,
    0,
    0,
    1500,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "food": Array [
    0,
    0,
    0,
    1500,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "ordersCount": Array [
    0,
    0,
    0,
    1,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "other": Array [
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "sum": Array [
    0,
    0,
    0,
    3000,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
}
`;

exports[`Dashboard pure function tests hourlyBreakdownOrderAmounts should calculate failed history items 1`] = `
Object {
  "drink": Array [
    0,
    0,
    0,
    30000,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "food": Array [
    0,
    0,
    0,
    30000,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "ordersCount": Array [
    0,
    0,
    0,
    20,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "other": Array [
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "sum": Array [
    0,
    0,
    0,
    60000,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
}
`;

exports[`Dashboard pure function tests hourlyBreakdownOrderAmounts should calculate success history items 1`] = `
Object {
  "drink": Array [
    0,
    0,
    0,
    4500,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "food": Array [
    0,
    0,
    0,
    4500,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "ordersCount": Array [
    0,
    0,
    0,
    3,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "other": Array [
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "sum": Array [
    0,
    0,
    0,
    9000,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
}
`;

exports[`Dashboard pure function tests unpayCategoryTableData should calculate hasIncome stats 1`] = `
Array [
  Object {
    "category": "delivery",
    "count": 2,
    "paymentMethodSums": Object {
      "card": 2982,
      "cash": 2982,
      "inapp": 0,
    },
    "sum": 5964,
    "uniqueUsersCount": 1,
  },
  Object {
    "category": "coupon",
    "count": 2,
    "paymentMethodSums": Object {
      "card": 2982,
      "cash": 2982,
      "inapp": 0,
    },
    "sum": 5964,
    "uniqueUsersCount": 1,
  },
  Object {
    "category": "event",
    "count": 2,
    "paymentMethodSums": Object {
      "card": 2982,
      "cash": 2982,
      "inapp": 0,
    },
    "sum": 5964,
    "uniqueUsersCount": 1,
  },
  Object {
    "category": "sum",
    "count": 6,
    "paymentMethodSums": Object {
      "card": 8946,
      "cash": 8946,
      "inapp": 0,
    },
    "sum": 17892,
    "uniqueUsersCount": 1,
  },
]
`;

exports[`Dashboard pure function tests unpayCategoryTableData should calculate hasIncome stats 2`] = `
Array [
  Object {
    "category": "staff_meal",
    "count": 2,
    "paymentMethodSums": Object {
      "card": 2982,
      "cash": 2982,
      "inapp": 0,
    },
    "sum": 5964,
    "uniqueUsersCount": 1,
  },
  Object {
    "category": "manager_meal",
    "count": 2,
    "paymentMethodSums": Object {
      "card": 2982,
      "cash": 2982,
      "inapp": 0,
    },
    "sum": 5964,
    "uniqueUsersCount": 1,
  },
  Object {
    "category": "marketing_promo",
    "count": 2,
    "paymentMethodSums": Object {
      "card": 2982,
      "cash": 2982,
      "inapp": 0,
    },
    "sum": 5964,
    "uniqueUsersCount": 1,
  },
  Object {
    "category": "error_cooked",
    "count": 2,
    "paymentMethodSums": Object {
      "card": 2982,
      "cash": 2982,
      "inapp": 0,
    },
    "sum": 5964,
    "uniqueUsersCount": 1,
  },
  Object {
    "category": "error_no_cooked",
    "count": 2,
    "paymentMethodSums": Object {
      "card": 2982,
      "cash": 2982,
      "inapp": 0,
    },
    "sum": 5964,
    "uniqueUsersCount": 1,
  },
  Object {
    "category": "payment_mode_change",
    "count": 2,
    "paymentMethodSums": Object {
      "card": 2982,
      "cash": 2982,
      "inapp": 0,
    },
    "sum": 5964,
    "uniqueUsersCount": 1,
  },
  Object {
    "category": "other",
    "count": 2,
    "paymentMethodSums": Object {
      "card": 2982,
      "cash": 2982,
      "inapp": 0,
    },
    "sum": 5964,
    "uniqueUsersCount": 1,
  },
  Object {
    "category": "sum",
    "count": 14,
    "paymentMethodSums": Object {
      "card": 20874,
      "cash": 20874,
      "inapp": 0,
    },
    "sum": 41748,
    "uniqueUsersCount": 1,
  },
]
`;
